### ISSUE TITLE
Either "cuDNN SDPA backward got grad_output.strides() != output.strides()" or OutOfMemory

### ISSUE THREAD

**Issue by Tablaski** [kohya-ss/sd-scripts#1609]
**TL;DR : very hard to tell if training is starting before getting first samples pics ; either it's just not starting or it fails with some error if trying other GPU external settings ; seems to work using 12 GB instead of 16 GB (for a 16 GB GPU)**

I'm confused... The part where the training start is a challenge
I have made fluxgym train 2 Loras and they came out well, but most of the time I get this :

[2024-09-17 09:01:52] [INFO] C:\fluxgym\env\Lib\site-packages\torch\autograd\graph.py:826: UserWarning: cuDNN SDPA backward got grad_output.strides() != output.strides(), attempting to materialize a grad_output with matching strides... (Triggered internally at C:\actions-runner_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\cudnn\MHA.cpp:676.)
[2024-09-17 09:01:52] [INFO] return Variable._execution_engine.run_backward( # Calls into the C++ engine to run the backward pass

Then it doesn't do anything. I've noticed sometimes it does work in the background, but you don't see it unless you have set up the parameter to have sample images regularly. So I set up a low threshold like 100 or 200 steps, this way I can monitor the training, because everytime it generates the sample images THEN you see the steps counters.

Unfortunately I just had luck because now I cannot train a third lora using the same parameters as the two previous. Here's what I tried so far :

Updated fluxgym to the latest version using git pull and refreshing dependencies. I ended up doing a fresh new install

Switched the NVIDIA 3D parameters in ControlPanel for CUDA Fallback policy to "Prefer No System Fallback" as someone recommended in https://github.com/cocktailpeanut/fluxgym/issues/48

This replaces the previous error by an OutOfMemory error where it says it's 64 MB short and I should maybe try to set PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True

Set PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
Now it says in the logs before it starts training that this setting is not allowed on this platform
Then I have a return with code 1

Switch VRAM to 12 Gb (instead of 16 GB)
Now this seems to work but it is a disappointing workaround as it seems much slower (first steps are taking around 7 s / step when I average 2.5 s / step with 16 GB)

Any idea how to fix that and/or how we could make the training start phase more reliable ?

There's this moment between the log line "return Variable._execution_engine.run_backward) and the first samples where I don't know whether or not it's gonna work

**Comment by Tablaski** [comment]
I get this error 

torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 72.00 MiB. GPU 0 has a total capacity of 15.99 GiB of which 0 bytes is free. Of the allocated memory 14.72 GiB is allocated by PyTorch, and 381.89 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)

I don't understand how it's not possible to take 72 MB out of 381.89 unallocated ?
Or it's because it's reserve ? Don't know what to do

Training with the 12 GB option is 3-4 times slower.

I'm using settings that worked for two Loras so when it wants, it can...

**Comment by Tablaski** [comment]
Retrying with the NVIDIA 3D parameters in ControlPanel for CUDA Fallback policy to "Automatic"

I get this

[2024-09-17 18:36:42] [INFO] C:\fluxgym\env\Lib\site-packages\torch\autograd\graph.py:826: UserWarning: cuDNN SDPA backward got grad_output.strides() != output.strides(), attempting to materialize a grad_output with matching strides... (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\cudnn\MHA.cpp:676.)
[2024-09-17 18:36:42] [INFO] return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass

I would say it is working due to the GPU temperature rising a bit, but it's the only indication I have after 20 minutes waiting
